File: python-sdks/examples/openai-agents-sdk/hello_world.py
"""Hello World — Minimal agent with Respan tracing."""

import asyncio
from dotenv import load_dotenv

load_dotenv(override=True)

from respan import Respan
from respan_instrumentation_openai_agents import OpenAIAgentsInstrumentor
from agents import Agent, Runner, trace

respan = Respan(instrumentations=[OpenAIAgentsInstrumentor()])

agent = Agent(
    name="Assistant",
    instructions="You only respond in haikus.",
)


async def main():
    with trace("Hello world"):
        result = await Runner.run(agent, "Tell me about recursion in programming.")
        print(result.final_output)
    respan.flush()


if __name__ == "__main__":
    asyncio.run(main())

File: javascript-sdks/examples/openai-agents-sdk/hello_world.ts
/** Hello World — Minimal agent with Respan tracing (mirrors Python example). */

import "dotenv/config";
import { Agent, run } from "@openai/agents";
import { Respan } from "@respan/respan";
import { OpenAIAgentsInstrumentor } from "@respan/instrumentation-openai-agents";

const respan = new Respan({
  apiKey: process.env.RESPAN_API_KEY,
  baseURL: process.env.RESPAN_BASE_URL,
  instrumentations: [new OpenAIAgentsInstrumentor()],
});
await respan.initialize();

const agent = new Agent({
  name: "Assistant",
  instructions: "You only respond in haikus.",
});

const result = await run(agent, "Tell me about recursion in programming.");
console.log(result.finalOutput);
